cenpy/product.py

```
@@ -0,0 +1,222 @@
+                result['GEOID'] = result['GEOID'].str.replace('R', '')
+                result['GEOID'] = result['GEOID'].str.replace('T', '')
+
+            if sdf.shape > (0, 0):
+                sdf = sdf.append(result, ignore_index=True)
+
+            else:
+                sdf = result
+
+        # merge dataframes
+        combined = pd.merge(df, sdf, on=['GEOID'])
+
+        # drop any duplicate GEOID (targetted to AIANNH)
+        combined = combined.loc[combined['GEOID'].drop_duplicates().index].reset_index(drop=True)
+
+        # reorder columns
+        outFields = outFields.split(',')
+        combined = combined[
+            geographies +
+            outFields +
+            [i for i in combined.columns if i not in geographies + outFields]
+        ]
+
+        if 'geometry' in combined.columns:
+            combined = gpd.GeoDataFrame(combined)
+
+        return combined
+
+
+class ACS(ProductBase):
+
+    def __init__(self, year, session=None):
+        super(ACS, self).__init__(year, session=session)
+        self._census = CensusDataset(
+            f'https://api.census.gov/data/{self.year}/acs/acs5',
+            session=self._session,
+        )
+        self._tiger = EsriMapServer(
+            f'https://tigerweb.geo.census.gov/arcgis/rest/services/TIGERweb/tigerWMS_ACS{self.year}/MapServer',
+            session=self._session,
+        )
+
+
+class Decennial(ProductBase):
+
+    def __init__(self, year, session=None):
+        super(Decennial, self).__init__(year, session=session)
+        self._census = CensusDataset(
+            f'https://api.census.gov/data/{self.year}/dec/sf1',
+            session=self._session,
+        )
+        self._tiger = EsriMapServer(
+            f'https://tigerweb.geo.census.gov/arcgis/rest/services/TIGERweb/tigerWMS_Census{self.year}/MapServer',
+            session=self._session,
+        )
+
+    @lazy_property
+    def _legislative_year(self):
+        return 'NotImpementedError'
```

Should these be `raise NotImplementedError()` rather than return the string?